nemo/lightning/pytorch/trainer.py

```diff
@@ -10,18 +10,23 @@
 
 
 class Trainer(pl.Trainer, IOMixin):
+
+    def add_io(self, obj):
```

Comment:
This function can be improved. You can use isinstance checks with the `Sequence` + `Mapping` types. I think it's easier to read if we start with that and move the top part into an else.